From 8f086fc9c59fdcfb516ca65f52a03036a787ee67 Mon Sep 17 00:00:00 2001
From: "Dragusu, Mihai" <Mihai.Dragusu@windriver.com>
Date: Thu, 19 Dec 2019 15:04:42 +0200
Subject: [PATCH] vxworks specific

undef READ
add yaml dependency

NOTE: temporary removal of add_sensors
and odometry publisher due to instabilities needs further investigation

Signed-off-by: Dragusu, Mihai <Mihai.Dragusu@windriver.com>
---
 turtlebot3_node/CMakeLists.txt                            | 4 +++-
 turtlebot3_node/include/turtlebot3_node/control_table.hpp | 3 +++
 turtlebot3_node/src/odometry.cpp                          | 3 ++-
 turtlebot3_node/src/turtlebot3.cpp                        | 2 ++
 4 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/turtlebot3_node/CMakeLists.txt b/turtlebot3_node/CMakeLists.txt
index 3f5c36b..cc6311f 100644
--- a/turtlebot3_node/CMakeLists.txt
+++ b/turtlebot3_node/CMakeLists.txt
@@ -28,6 +28,8 @@ find_package(std_srvs REQUIRED)
 find_package(tf2 REQUIRED)
 find_package(tf2_ros REQUIRED)
 find_package(turtlebot3_msgs REQUIRED)
+find_package(libyaml_vendor REQUIRED)
+find_package(yaml REQUIRED)
 
 ################################################################################
 # Build
@@ -73,7 +75,7 @@ ament_target_dependencies(${PROJECT_NAME}_lib ${DEPENDENCIES})
 set(EXECUTABLE_NAME "turtlebot3_ros")
 
 add_executable(${EXECUTABLE_NAME} src/node_main.cpp)
-target_link_libraries(${EXECUTABLE_NAME} ${PROJECT_NAME}_lib)
+target_link_libraries(${EXECUTABLE_NAME} ${PROJECT_NAME}_lib "yaml")
 ament_target_dependencies(${EXECUTABLE_NAME} ${DEPENDENCIES})
 
 ################################################################################
diff --git a/turtlebot3_node/include/turtlebot3_node/control_table.hpp b/turtlebot3_node/include/turtlebot3_node/control_table.hpp
index 55130b8..87c83d3 100644
--- a/turtlebot3_node/include/turtlebot3_node/control_table.hpp
+++ b/turtlebot3_node/include/turtlebot3_node/control_table.hpp
@@ -28,6 +28,9 @@ namespace turtlebot3
 constexpr uint8_t EEPROM = 1;
 constexpr uint8_t RAM = 2;
 
+#ifdef READ
+#undef READ
+#endif
 constexpr uint8_t READ = 1;
 constexpr uint8_t READ_WRITE = 3;
 
diff --git a/turtlebot3_node/src/odometry.cpp b/turtlebot3_node/src/odometry.cpp
index 898cb59..d926844 100644
--- a/turtlebot3_node/src/odometry.cpp
+++ b/turtlebot3_node/src/odometry.cpp
@@ -62,8 +62,9 @@ Odometry::Odometry(
     std::string("base_footprint"));
 
   auto qos = rclcpp::QoS(rclcpp::KeepLast(10));
+#ifndef __VXWORKS__
   odom_pub_ = nh_->create_publisher<nav_msgs::msg::Odometry>("odom", qos);
-
+#endif
   tf_broadcaster_ = std::make_unique<tf2_ros::TransformBroadcaster>(nh_);
 
   if (use_imu_)
diff --git a/turtlebot3_node/src/turtlebot3.cpp b/turtlebot3_node/src/turtlebot3.cpp
index f714819..1e97458 100644
--- a/turtlebot3_node/src/turtlebot3.cpp
+++ b/turtlebot3_node/src/turtlebot3.cpp
@@ -33,7 +33,9 @@ TurtleBot3::TurtleBot3(const std::string & usb_port)
 
   add_motors();
   add_wheels();
+#ifndef __VXWORKS__
   add_sensors();
+#endif
   add_devices();
 
   run();
-- 
2.24.0

